code/modules/hydroponics/trays/tray_process.dm

```diff
@@ -20,8 +20,9 @@
 
 	// Weeds like water and nutrients, there's a chance the weed population will increase.
 	// Bonus chance if the tray is unoccupied.
-	if(waterlevel > 10 && nutrilevel > 2 && prob(isnull(seed) ? 5 : 1))
-		weedlevel += 1 * HYDRO_SPEED_MULTIPLIER
+	if(!mechanical) // Changes it so that only soil plots need to worry about weeds.
```

Comment:
Farmbot checks weedlevel var. So now with only pestlevel farmbot will not clean mechanical trays from pest.s